Add page regarding content guidelines (#339)
precice/precice.github.io: pages/docs/docs-meta/docs-meta-content-guidelines.md

---
title: Content guidelines
permalink: docs-meta-content-guidelines.html

> keywords: pages, migration, guideline, content, best practises

Reviewer: ```suggestion
keywords: pages, migration, guideline, content, best practices, style guide
```

summary: "Learn which style to follow when writing documentation and to choose good titles, content and page summaries."
---

## Language & style

As we recently (December 2020) migrated our documentation from multiple sources to this website, you may find different styles and inconsistencies among different pages. However, here is what we aim for:

- Target group: scientists & engineers with some but limited experience with programming and with Linux, but extended experience with simulations.
- Informal style and active voice: imagine you are explaining each concept to a colleague over coffee.
- Concise, yet complete: short pages are completely fine and even preferred, as long as all the important information is there.
- Incomplete/imperfect documentation is better than no documentation: try to contribute anything you can, and we can always improve it.
- We use `Sentence case for headings`, not `Title Case for Headings`. The reason is that we find that it is visually clearer, easier to keep it consistent, and we do not need to mix content with style.

> - Descriptive links: avoid forms such as `you can find the documentation [here](target)`, prefer forms such as `see the [documentation](target)`.

Reviewer: ```suggestion
- Descriptive links: avoid forms such as `you can find the documentation [here](target)`, prefer forms such as `see the [documentation](target)`.
- We use American English.
```

---
title: Content guidelines
permalink: docs-meta-content-guidelines.html
keywords: pages, migration, guideline, content, best practices, style guide

> summary: "Learn which style to follow when writing documentation and to choose good titles, content and page summaries."

Reviewer 2: ```suggestion
summary: "Learn which style to follow when writing documentation and how to write good titles, content, and page summaries."
```

---

## Language & style

> As we recently (December 2020) migrated our documentation from multiple sources to this website, you may find different styles and inconsistencies among different pages. However, here is what we aim for:

Reviewer 2: ```suggestion
This is the style we aim for, even if not all documentation pages are currently ticking all the boxes (nice opportunity for [contributing](community-contribute-to-precice.html)):
```

- We use `Sentence case for headings`, not `Title Case for Headings`. The reason is that we find that it is visually clearer, easier to keep it consistent, and we do not need to mix content with style.
- Descriptive links: avoid forms such as `you can find the documentation [here](target)`, prefer forms such as `see the [documentation](target)`.
- We use American English.

## Titles

- Always specify a `title:`
- Keep it short, specific, and avoid boilerplate.
- Avoid duplicate titles.
- Don't include navigational context such as `preCICE.org |`, `preCICE Documentaion -`. Jekyll does this automatically.

More information on the [title link guide from Google](https://developers.google.com/search/docs/appearance/title-link#page-titles).

## Page summary

- Always specify a `summary:`
- This text may be chosen as the preview text under a search result.
- Write a flowing text. Multiple sentences are fine.
- Summarize the whole page and be specific.
- Include important information like dates.

See full [snippet and meta-description guide from Google](https://developers.google.com/search/docs/appearance/snippet#meta-descriptions)

## Images

> - Always specify a textual description using the full Markdown syntax: `![DESCRIPTION](LINK TO IMAGE)`

Reviewer 2: ```suggestion
- Always specify a textual description using the full Markdown syntax: `![DESCRIPTION](LINK TO IMAGE)`. This is important both for crawlers and for screen readers.
```

> - Stick to common image formats BMP, GIF, JPEG, PNG, WebP, and SVG

Reviewer 2: ```suggestion
- Stick to common image formats, including PNG, WebP, GIF, and SVG. In case of SVG files, make sure that text is rendered properly outside your browser (e.g., by converting text to paths). Other supported (but discouraged) formats include JPEG and BMP.
```

Author: If we want to make recommendations, then we can be more specific:

```suggestion
- For rasterized graphics and animations use [`WebP`](https://repology.org/project/libwebp/versions). Convert images using [`cwebp`](https://developers.google.com/speed/webp/docs/cwebp) and create animations using [`img2webp`](https://developers.google.com/speed/webp/docs/img2webp).
- For vectorized graphics use SVG. Make sure text renders correctly outside your browser by using common fonts or converting text to paths. In case of the latter, also add the original SVG to simplify future modifications.
```

Reviewer 2: I would still mention PNG: Most of our images are currently PNG, and people should feel welcome to submit such.

Author: This documentation is primarily meant for preCICE devs, right? This is always something we can manually do for submission.

Reviewer 2: I would still show this page to people that want to contribute documentation, but I don't see why this is important. In any case, this is very minor.

Author: Added PNG and split animations into a separate point.

```suggestion
- For rasterized graphics use PNG or [`WebP`](https://repology.org/project/libwebp/versions). For the latter, convert images using [`img2webp`](https://developers.google.com/speed/webp/docs/img2webp) with `img2webp image.png -o image.webp`.
- For animations use [`WebP`](https://repology.org/project/libwebp/versions). Convert exisiting GIFs using [`gif2webp`](https://developers.google.com/speed/webp/docs/gif2webp) or create new animations with `ffmpeg -i image%d.png -loop 0 output.webp`.
- For vectorized graphics use SVG. Make sure text renders correctly outside your browser by using common fonts or converting text to paths. In case of the latter, also add the original SVG to simplify future modifications.
```

- Choose a descriptive filename

See full [image best practices from Google](https://developers.google.com/search/docs/appearance/google-images).

## General content

- Avoid boilerplate content and get to the point as quickly as possible.
- If content already exists on another side, prefer linking to it unless you can add some value.

> - Don't copy external content.

Reviewer 2: ```suggestion
- Don't copy external content, unless you are making a summary of complicated external resources (e.g., installation instructions).
```

Author: That's what I mean with `prefer linking to it unless you can add some value`.

Maybe we need to merge both points to something like this:

> If content already exists on another side, prefer linking to it. Simplifying, customizing, or summarizing external content is fine. Don't copy chunks of documentation.

Reviewer 2: I think the important part is `Don't copy text without written permission.`.
